nautobot/ipam/tests/test_api.py

```diff
@@ -154,25 +163,45 @@ def setUpTestData(cls):
             virtual_machine=cls.test_vm,
             rd="65000:4",
         )
+        VRFDeviceAssignment.objects.create(
+            vrf=cls.vrfs[0],
+            virtual_device_context=cls.vdcs[0],
+            rd="65000:5",
+        )
+        VRFDeviceAssignment.objects.create(
+            vrf=cls.vrfs[0],
+            virtual_device_context=cls.vdcs[1],
+            rd="65000:6",
+        )
 
         cls.create_data = [
             {
                 "vrf": cls.vrfs[2].pk,
                 "device": cls.devices[4].pk,
                 "virtual_machine": None,
-                "rd": "65000:4",
+                "virtual_device_context": None,
+                "rd": "65000:7",
             },
             {
                 "vrf": cls.vrfs[3].pk,
                 "device": None,
                 "virtual_machine": cls.test_vm.pk,
-                "rd": "65000:5",
+                "virtual_device_context": None,
+                "rd": "65000:8",
             },
             {
                 "vrf": cls.vrfs[4].pk,
                 "device": cls.devices[6].pk,
                 "virtual_machine": None,
-                "rd": "65000:6",
+                "virtual_device_context": None,
+                "rd": "65000:9",
+            },
+            {
+                "vrf": cls.vrfs[4].pk,
+                "device": None,
+                "virtual_machine": None,
+                "virtual_device_context": cls.vdcs[0].pk,
+                "rd": "65000:10",
```

Comment:
Here too let's make sure the test covers both providing and omitting the fields (name, rd) that should be optional.